packages/ui/src/components/va-modal/VaModal.vue

```diff
@@ -270,6 +291,16 @@ export default defineComponent({
           document.value?.body.classList.remove('va-modal-overlay-background--blurred')
         }
       }
+
+      if (isTopLevelModal.value && modalDialog.value) {
```

Comment:
I think it must be moved to its own `watchEffect`. It looks like it will add or remove window events if `isTopLevelModal` changed. Same with css classes. It will cause a bug with blurred body if we close-open nested modal.

Reply:
Moved to it's own `wathcEffect`